### PR TITLE
Added new gRPC method to fetch system observability data

### PR DESCRIPTION
**Pull request by seswarrajan**
Signed-off-by: Eswar Rajan Subramanian <eswar@accuknox.com>

### CODE REVIEW

File: scripts/get_sys_observability_data.sh [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
```
Comment (nyrahul) [inline scripts/get_sys_observability_data.sh:1]: we don't want the input to be provided in this form (i.e., do not use stdin to get input)...

Read the parameters from commandline...

`./get_sys_observability_data.sh --cluster default --container xyz --namespace abc --labels "xyz=abc, lmn=pqr"`

File: scripts/get_sys_observability_data.sh [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+echo "Input containername"
+read containername
+
+echo "Input namespace"
+read namespace
+
+echo "Input labels"
+read labels
+
+DATA='{"clusterName": "'$clustername'", "namespace":"'$namespace'", "containerName":"'$containername'", "labels":"'$labels'"}'
```
Comment (nyrahul) [inline scripts/get_sys_observability_data.sh:20]: Is it necessary that all the parameters are provided?

If no clustername is provided .. then we should be using all clusters ... Similarly for other fields.

File: src/libs/common.go [hunk 3]
```diff
@@ -452,6 +462,34 @@ func WriteKubeArmorPolicyToYamlFile(fname string, namespace string, policies []t
 	}
 }
 
+func WriteSysObsDataToJsonFile(obsData types.SysObservabilityData) {
```
Comment (nyrahul) [inline src/libs/common.go:465]: Send the output as response to request ... Not dump it in the file.

File: src/systempolicy/observability.go [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+		return nil
+	}
+
+	var sysObsProcessFileData []types.SysObsProcessFileData
+
+	for wpfs, fsset := range wpfsSet {
+		var locSysObsProcessFileData types.SysObsProcessFileData
+
+		locSysObsProcessFileData.FromSource = wpfs.FromSource
+		if wpfs.SetType == "Process" {
```
Comment (nyrahul) [inline src/systempolicy/observability.go]:
```suggestion
		if wpfs.SetType == SYS_OP_PROCESS {
```

File: src/systempolicy/observability.go [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+			locSysObsProcessFileData.ProcessPaths = append(locSysObsProcessFileData.ProcessPaths, fsset...)
+		}
+		if wpfs.SetType == "File" {
```
Comment (nyrahul) [inline src/systempolicy/observability.go]:
```suggestion
		if wpfs.SetType == SYS_OP_FILE {
```

File: src/systempolicy/observability.go [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+	wpfs.Labels = labels
+
+	res, policyNames, err := libs.GetWorkloadProcessFileSet(CfgDB, wpfs)
+	if err != nil {
+		return err
+	}
+
+	sysObsProcessFileData := convertWPFSToObservabilityData(res, policyNames)
+	if len(sysObsProcessFileData) == 0 {
+		return errors.New("no data in db")
```
Comment (nyrahul) [inline src/systempolicy/observability.go]: This should not be an error. Ok to have no data for the given filters in the DB.

**Review by nyrahul — CHANGES_REQUESTED** [review]
please fix the new lint issues.

File: scripts/get_sys_observability_data.sh (hunk 2 above)
Comment (seswarrajan) [reply scripts/get_sys_observability_data.sh:20]: Yes Rahul. If we did not provide cluster name, all clusters will be read. If any vallue is ignored then * is used in query

**Review by nyrahul — APPROVED** [review]

**Pull request merged**